internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -739,6 +765,65 @@ func (r *BMACReconciler) ensureSpokeMachine(ctx context.Context, spokeClient cli
 	return machineSpoke, nil
 }
 
+// Get MCS cert, encode it and add it to BMH which will add ultimately into agent
+func (r *BMACReconciler) ensureMCSCert(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost) error {
+	ignitionWithMCSCert, err := r.createIgnitionWithMCSCert(ctx)
+	if err != nil {
+		return err
+	}
+	bmhAnnotations := bmh.ObjectMeta.GetAnnotations()
+	userIgnition, ok := bmhAnnotations[BMH_AGENT_IGNITION_CONFIG_OVERRIDES]
+	if bmh.ObjectMeta.Annotations == nil {
+		bmh.ObjectMeta.Annotations = make(map[string]string)
+	}
+	// User has set ignition via annotation
+	if ok {
+		// Here we have 2 cases
+		// case 1: security.tls.certificateAuthorities array exists in the user provided ignition
+		// case 2: security.tls.certificateAuthorities array doesn't exist in the user provided ignition
+		res, err := ignition.MergeIgnitionConfig([]byte(userIgnition), []byte(ignitionWithMCSCert))
```

Comment:
I am glad there is a Merge function already and that we don't have to re-implement it. I was not aware :)